File: project/app/src/main/java/org/owntracks/android/injection/modules/ActivityModules/ConnectionActivityModule.java
package org.owntracks.android.injection.modules.ActivityModules;

import android.support.v7.app.AppCompatActivity;

import org.owntracks.android.injection.modules.ActivityModule;
import org.owntracks.android.injection.qualifier.ActivityContext;
import org.owntracks.android.injection.scopes.PerActivity;
import org.owntracks.android.ui.map.MapMvvm;
import org.owntracks.android.ui.preferences.connection.ConnectionActivity;
import org.owntracks.android.ui.preferences.connection.ConnectionMvvm;
import org.owntracks.android.ui.preferences.connection.ConnectionViewModel;

import dagger.Binds;
import dagger.Module;

@Module(includes = ActivityModule.class)
public abstract class ConnectionActivityModule {

    @Binds
    @PerActivity
    abstract AppCompatActivity appCompatActivity(ConnectionActivity a);

    @Binds abstract ConnectionMvvm.ViewModel bindViewModel(ConnectionViewModel viewModel);
}

File: project/app/src/main/java/org/owntracks/android/ui/preferences/connection/ConnectionActivityModule.kt
package org.owntracks.android.ui.preferences.connection

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@InstallIn(ActivityComponent::class)
@Module
abstract class ConnectionActivityModule {
    @Binds
    abstract fun bindViewModel(viewModel: ConnectionViewModel): ConnectionMvvm.ViewModel<ConnectionMvvm.View>
}
